pipenv/core.py

```diff
@@ -1367,6 +1367,7 @@ def pip_install(
     pre=False,
     selective_upgrade=False,
     requirements_dir=None,
+    extra_indexes=None,
```

Comment:
wdyt about, rather than making this type polymorphic, enforcing that `extra_indexes` must always be a list and  when calling this function on line 841, ensuring that it's always a list (or having split_extra_index always return a list of elements)

Reply:
I think that makes sense but I want to stay out of the weeds on that logic until I get this bugfix released, so lets plan for 11.11.0